sympy/solvers/ode/ode.py

```diff
@@ -6686,7 +6686,7 @@ def _nonlinear_2eq_order1_type3(x, y, t, eq):
     F = r1[f].subs(x(t), u).subs(y(t), v(u))
     G = r2[g].subs(x(t), u).subs(y(t), v(u))
     sol2r = dsolve(Eq(diff(v(u), u), G/F))
-    if isinstance(sol2r, Expr):
+    if isinstance(sol2r, Equality):
```

Comment:
I didn't check this thoroughly before. Is it guaranteed that sol2r will always be either an Equality or a list of Equality? 

Equations can be provided as Expr: would they be converted to Equality before this point?

Reply:
Hello thanks for the review.

I looked at it and realized that there are actually no guarantees that sol2r will be an Equality.

Should we make a check on sol2r and make it an Equality if it is not? For instance could we equate to 0?

Reply:
Do you have an example of inputs to `dsolve` that would not work with the change you have made here?

Reply:
Nope. I dont have inputs that will not work given this change.

Reply:
I think earlier in dsolve the inputs are converted to Equality if they are not already. Probably this bug came because previously Equality was a subclass of Expr but I changed that at some point. Since this code was untested this case was not picked up on at the time.

That's why it's important to have tests!

Thanks for this.

Reply:
Understood. Thanks for the guidance!